src/model.d.ts

```diff
@@ -2762,12 +2767,19 @@ export abstract class Model<TModelAttributes extends {} = any, TCreationAttribut
   /**
    * Deletes multiple instances, or set their deletedAt timestamp to the current time if `paranoid` is enabled.
    *
-   * @returns The number of destroyed rows
+   * The promise resolves with an array of one or two elements
+   * - The first element is always the number of destroyed rows
+   * - The second element is the list of destroyed entities (only supported in postgres)
    */
+  static destroy<M extends Model>(
+    this: ModelStatic<M>,
+    options: Omit<DestroyOptions<Attributes<M>>, 'returning'>
+      & { returning: Exclude<DestroyOptions<Attributes<M>>['returning'], undefined | false> }
+  ): Promise<[deletedCount: number, deletedRows: M[]]>;
   static destroy<M extends Model>(
     this: ModelStatic<M>,
     options?: DestroyOptions<Attributes<M>>
-  ): Promise<number>;
```

Comment:
Can we keep this as a number? I think that's a nicer solution.

Reply:
Yeah, I was 50/50 on that. I was hoping to get a review on that. Will keep it as a number :+1: